Some tests are failing

It's April 1st and time for the traditional "some tests fail because it's a new year" failures:

https://github.com/rcpch/national-paediatric-diabetes-audit/actions/runs/23843578167/job/69506498544?pr=1367

```
FAILED project/npda/tests/factories/test_fake_patient_builder.py::test_fake_patient_creator_ages_all_appropriate[AgeRange.AGE_0_4]
FAILED project/npda/tests/factories/test_fake_patient_builder.py::test_fake_patient_creator_ages_all_appropriate[AgeRange.AGE_16_19]
FAILED project/npda/tests/form_tests/test_visit_form.py::test_smoking_status_smoker_does_not_require_cessation_referral_date
FAILED project/npda/tests/permissions_tests/test_npda_user_model_actions.py::test_users_can_download_csv[user_data0]
FAILED project/npda/tests/permissions_tests/test_npda_user_model_actions.py::test_users_can_download_csv[user_data1]
FAILED project/npda/tests/permissions_tests/test_npda_user_model_actions.py::test_users_can_download_csv[user_data2]
FAILED project/npda/tests/permissions_tests/test_npda_user_model_actions.py::test_users_can_download_report[user_data0]
FAILED project/npda/tests/permissions_tests/test_npda_user_model_actions.py::test_users_can_download_report[user_data1]
FAILED project/npda/tests/permissions_tests/test_npda_user_model_actions.py::test_users_can_download_report[user_data2]
FAILED project/npda/tests/permissions_tests/test_npda_user_model_actions.py::test_rcpch_audit_team_can_delete_submission
FAILED project/npda/tests/view_tests/test_patient_report.py::test_no_duplicate_patients_in_report
```

--- a/project/npda/general_functions/data_generator_extended.py
+++ b/project/npda/general_functions/data_generator_extended.py
@@ -233,6 +233,8 @@ class FakePatientCreator:
             patients = PatientFactory.create_batch(
                 size=n,
                 age_range=age_range,
+                audit_start_date=self.audit_start_date,
+                audit_end_date=self.audit_end_date,
                 # We're going to manually create visits for each patient
                 visit=None,
                 **patient_kwargs,

--- a/project/npda/tests/factories/seed_audit_period.py
+++ b/project/npda/tests/factories/seed_audit_period.py
@@ -21,36 +21,20 @@ def _seed_audit_periods_fixture(django_db_setup, django_db_blocker):
         current_audit_year = current_audit_start.year
 
         if AuditPeriod.objects.exists():
-            # DB was reused — ensure the current audit year exists even if it wasn't seeded before
-            if not AuditPeriod.objects.filter(
-                slug=f"{current_audit_year}-{current_audit_year + 1}"
-            ).exists():
-                logger.info(
-                    f"Seeding missing current audit period {current_audit_year}-{current_audit_year + 1}."
-                )
+            logger.info("NOTE: Test audit periods already seeded! Not re-seeding.")
+        else:
+            for start_year in range(2024, current_audit_year + 1):
+                end_year = start_year + 1
+
+                logger.info(f"Seeding test audit period {start_year}-{end_year}.")
+
                 AuditPeriod.objects.create(
                     is_open=True,
                     is_visible=True,
-                    start_date=f"{current_audit_year}-04-01",
-                    end_date=f"{current_audit_year + 1}-03-31",
-                    slug=f"{current_audit_year}-{current_audit_year + 1}",
+                    start_date=f"{start_year}-04-01",
+                    end_date=f"{end_year}-03-31",
+                    slug=f"{start_year}-{end_year}",
                 )
-            else:
-                logger.info("NOTE: Test audit periods already seeded! Not re-seeding.")
-            return
-
-        for start_year in range(2024, current_audit_year + 1):
-            end_year = start_year + 1
-
-            logger.info(f"Seeding test audit period {start_year}-{end_year}.")
-
-            AuditPeriod.objects.create(
-                is_open=True,
-                is_visible=True,
-                start_date=f"{start_year}-04-01",
-                end_date=f"{end_year}-03-31",
-                slug=f"{start_year}-{end_year}",
-            )
 
 
 @pytest.fixture(scope="session")

--- a/project/npda/tests/factories/seed_audit_period.py
+++ b/project/npda/tests/factories/seed_audit_period.py
@@ -3,11 +3,13 @@ Seeds NPDA Users in test db once per session.
 """
 
 import logging
+from datetime import date
 
 # Standard imports
 import pytest
 
 # NPDA Imports
+from project.npda.general_functions.audit_period import get_audit_period_for_date
 from project.npda.models import AuditPeriod
 
 logger = logging.getLogger(__name__)
@@ -15,11 +17,29 @@ logger = logging.getLogger(__name__)
 
 def _seed_audit_periods_fixture(django_db_setup, django_db_blocker):
     with django_db_blocker.unblock():
+        current_audit_start, _ = get_audit_period_for_date(date.today())
+        current_audit_year = current_audit_start.year
+
         if AuditPeriod.objects.exists():
-            logger.info("NOTE: Test audit periods already seeded! Not re-seeding.")
+            # DB was reused — ensure the current audit year exists even if it wasn't seeded before
+            if not AuditPeriod.objects.filter(
+                slug=f"{current_audit_year}-{current_audit_year + 1}"
+            ).exists():
+                logger.info(
+                    f"Seeding missing current audit period {current_audit_year}-{current_audit_year + 1}."
+                )
+                AuditPeriod.objects.create(
+                    is_open=True,
+                    is_visible=True,
+                    start_date=f"{current_audit_year}-04-01",
+                    end_date=f"{current_audit_year + 1}-03-31",
+                    slug=f"{current_audit_year}-{current_audit_year + 1}",
+                )
+            else:
+                logger.info("NOTE: Test audit periods already seeded! Not re-seeding.")
             return
 
-        for start_year in [2024, 2025]:
+        for start_year in range(2024, current_audit_year + 1):
             end_year = start_year + 1
 
             logger.info(f"Seeding test audit period {start_year}-{end_year}.")

